ping/ping_common.c

```diff
@@ -530,7 +531,7 @@ void setup(struct ping_rts *rts, socket_st *sock)
 			*p++ = i;
 	}
 
-	if (sock->socktype == SOCK_RAW)
+	if (sock->socktype == SOCK_RAW && rts->ident == -1)
 		rts->ident = rand() & 0xFFFF;
```

Comment:
Maybe use the constant when you defined it?
`rts->ident = rand() & IDENTIFIER_MAX;`

Reply:
done